Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            int numberOfIndividuals = PromptInt("Please enter the number of individuals you would like to calculate the BMI of: ");
+            while (numberOfIndividuals > 0)
+            {
+                PromptForPersonalData();
+
+                numberOfIndividuals--;
+            }
+        }
+
+        static void PromptForPersonalData()
+        {
+            string firstName = PromptString("Please enter the individual's first name: ");
+            string lastName = PromptString("Please enter the individual's last name: ");
+            int age = PromptInt("Please enter the individual's age: ");
+            float weight = PromptFloat("Please enter the individual's weight (in kg, example: 70.5): ");
+            float height = PromptFloat("Please enter the individual's height (in meters, example: 1.8): ");
+            float bodyMassIndex = CalculateBMI(weight, height);
+
+            Console.WriteLine($"{firstName} {lastName} is {age} years old, their weight is {weight} kg and their height is {height} m.");
+            Console.WriteLine($"{firstName} {lastName}'s BMI is: {bodyMassIndex}");
+        }
+
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateString(Console.ReadLine());
+        }
+
+        public static int PromptInt(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateInt(Console.ReadLine());
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return ValidateFloat(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float height)
+        {
+            if (weight <= 0 || height <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+                if (weight <= 0 && height <= 0)
+                {
+                    Console.WriteLine("Weight cannot be equal or less than zero, but was {0}.", weight);
+                    Console.WriteLine("Height cannot be less than zero, but was {0}.", height);
```

Comment:
Well done for spotting this silly inconsistency. It should be consistent, but then the tests fails. It's my bad that I will fix eventually. Anyways, well done.

Reply:
I was scratching my head for at least 10-15 minutes but at least I got to know more about the tests you wrote :)